pkg/controller/deployment/deployment_controller.go

```diff
@@ -107,5 +107,8 @@ func (r *ReconcileDeployment) Reconcile(request reconcile.Request) (reconcile.Re
 		return reconcile.Result{}, err
 	}
 
-	return r.handler.HandleDeployment(instance)
+	// Create a new PodController with the fetched deployment instance
+	podControllerInstance := &core.Deployment{Deployment: instance}
```

Comment:
I originally thought about having a wrapper for this
```
func (h *Handler) HandleDeployment(instance *appsv1.Deployment) (...) {
    return h.HandlePodController(&core.Deployment{Deployment: instance})
}
```

to abstract the `podController` stuff away from this package but I'm happy with this too

Reply:
Yeah that makes perfect sense, the new commit has wrapped the handle pod controller.